README.md

```diff
@@ -1,85 +1,73 @@
+### PVCBackupItemAction
 
-## Building the plugins
+A BackupItemAction plugin that backs up `persistentvolumeclaims` that are backed by a CSI volumes. 
 
-To build the plugins, run
+This plugin will create a [CSI VolumeSnapshot][3] which in turn trigger the CSI driver to perform the snapshot operation on the volume.
 
-```bash
-$ make
-```
+### VolumeSnapshotBackupItemAction
 
-## Known shortcomings
+A BackupItemAction plugin that backs up [`volumesnapshots.snapshot.storage.k8s.io`][3].
 
-* Deleting a backup doesn't clean up associated VolumeSnapshot(Contents) objects.
-* VolumeSnapshots taken with a backup aren't listed in the `backup describe` command.
-* VSLs must be deleted, otherwise you'll get PartiallyFailed status on your backup as the volume snapshot plugins will try to run and fail due to missing configuration for the storage class.
-* There is no VolumeSnapshot to VolumeSnapshotContent logic now, so to back up a namespace, you must back up all cluster resources.
-* Restic doesn't work for CSI-created volumes currently, as they have an additional directory that we don't expect.
-* `restore describe --detail` output needs to account for VolumeSnapshots
-* Deleting a VolumeSnapshotContent object _also deletes the underlying volume on the cloud provider_. This needs to be addressed in order for the full restoration process to work as it does with Velero.
+When invoked, this plugin will capture information about the underlying [`volumesnapshotcontent.snapshot.storage.k8s.io`][4] in the annotations of the volumesnapshots being backed up. This plugin will also return the underlying [`volumesnapshotcontent.snapshot.storage.k8s.io`][4] and the associated [`snapshot.storage.k8s.io.volumesnapshotclasses`][5] as additional resources to be backed up.
 
-## Using with AWS's EBS CSI driver
+### VolumeSnapshotContentBackupItemAction
 
-Deploy the driver (see [their docs](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/) for more details)
+A BackupItemAction plugin that backs up [`volumesnapshotcontent.snapshot.storage.k8s.io`][4]. 
 
-```
-kubectl apply -k "github.com/kubernetes-sigs/aws-ebs-csi-driver/deploy/kubernetes/overlays/stable/?ref=master"
-```
+This plugin will look for snapshot delete operation secrets from the [annotations][6] on the VolumeSnapshotContent object being backed up.
 
-Apply the storage class
+### VolumeSnapshotClassBackupItemAction
 
-```
-kubectl apply -f examples/sc.yaml
-```
+A BackupItemAction plugin that backs up [`snapshot.storage.k8s.io.volumesnapshotclasses`][5].
 
-Apply the snapshot class
+This plugin will look for snapshot list operation secret from the [annotations][6] on the VolumeSnapshotClass object being backed up.
 
-```
-kubectl apply -f examples/snapshotclass.yaml
-```
+### PVCRestoreItemAction
 
-Create the demo application
+A RestoreItemAction plugin that restores `persistentvolumeclaims` that were backed up by [PVCBackupItemAction](#PVCBackupItemAction).
```

Comment:
```suggestion
A RestoreItemAction plugin type that restores `persistentvolumeclaims` which were backed up by [PVCBackupItemAction](#PVCBackupItemAction).
```

Reply:
To note: pls double check the usage of present tense vs future tense. Some of these are using `will`, some are using `were`. I think if it all used present tense (ex "this plugin modified") it would be best, imo.

Reply:
Yes, the tenses changed bc while describing a restore, I am referring to what was done during backup. I'll double check though.